src/test/lib/HandshakeTest.cpp

```diff
@@ -4099,3 +4099,191 @@ QuicTestHandshakeSpecificLossPatterns(
+            TEST_NOT_EQUAL(nullptr, CibirIdBuffer);
+            QUIC_CONNECTION_POOL_CONFIG PoolConfig{};
+            PoolConfig.Registration = Registration;
+            PoolConfig.Configuration = ClientConfiguration;
+            PoolConfig.Handler = ConnectionPoolConnectionContext::ConnCallback;
+            PoolConfig.ServerName = QUIC_LOCALHOST_FOR_AF(QuicAddrFamily);
+            if (XdpSupported) {
+                PoolConfig.ServerAddress = &ServerAddr.SockAddr;
+            }
+            PoolConfig.ServerPort = ServerPort;
+            PoolConfig.Context = (void**)ContextPtrs.get();
+            PoolConfig.Family = QuicAddrFamily;
+            PoolConfig.NumberOfConnections = NumberOfConnections;
+            PoolConfig.Flags = QUIC_CONNECTION_POOL_FLAG_NONE;
+
+            if (TestCibirSupport) {
+                for (uint32_t i = 0; i < NumberOfConnections; i++) {
+                    CxPlatCopyMemory(&(CibirIdBuffer[i * CibirIdLength]), CibirId, CibirIdLength);
+                    CibirIdBuffer[(i * CibirIdLength) + (CibirIdLength - 1)] += (uint8_t)i;
+                    CibirIdPtrs[i] = &CibirIdBuffer[i * CibirIdLength];
+                }
+                PoolConfig.CibirIds = CibirIdPtrs.get();
+                PoolConfig.CibirIdLength = CibirIdLength;
+            }
+
+            QUIC_STATUS Status = MsQuic->ConnectionPoolCreate(&PoolConfig, &(Connections.get()->Handle));
+            if (XdpSupported) {
+                TEST_QUIC_SUCCEEDED(Status);
+                ServerAcceptContext.StartedEvent.WaitForever();
+                for (uint32_t i = 0; i < NumberOfConnections; i++) {
+                    //
+                    // Verify the server connections are connected. The client connections should also be connected too.
+                    //
+                    ServerConnections[i]->WaitForConnectionComplete();
+                    if (!ServerConnections[i]->GetIsConnected()) {
+                        TEST_FAILURE("Server connection %u failed to connect", i);
+                    }
+                    //
+                    // Verify the client connection is connected.
+                    //
+                    if (!Contexts[i].Connected) {
+                        TEST_FAILURE("Client connection %u failed to connect", i);
+                    }
+                }
+            } else {
+                //
+                // When testing no XDP support, the loopback address is used,
+                // and XDP doesn't support the loopback interface, so the
+                // expected error is "NOT_FOUND" since XDP isn't found for that
+                // interface.
+                // In the case the platform doesn't support XDP at all,
+                // NOT_SUPPORTED is expected.
+                //
+                if (Status != QUIC_STATUS_NOT_SUPPORTED &&
+#if defined(_WIN32) && !defined(_KERNEL_MODE)
+                    Status != QUIC_STATUS_FILE_NOT_FOUND &&
+#endif
```

Comment:
Maybe just define this on Posix to be the same thing as `QUIC_STATUS_NOT_FOUND` so you don't need this check?